lib/rules/require-v-if-else-key.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @author Doug Wade <douglas.b.wade@gmail.com>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+// ...
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'require `key` with `v-if/v-else-if/v-else` directives',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/require-v-if-else-key.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected:
+        'Elements in v-if/v-else-if/v-else expect to have distinct keys if they are of the same type.'
```

Comment:
Maybe _tag name_ is a bit clearer than _element type_?


```suggestion
        'Elements in v-if/v-else-if/v-else should have distinct keys if they have the same tag name.'
```